src/Tools/BuildBoss/ProjectUtil.cs

```diff
@@ -136,13 +136,28 @@ internal IEnumerable<XElement> GetItemGroup()
             return _document.XPathSelectElements("//mb:ItemGroup", _manager);
         }
 
-        internal List<ProjectKey> GetDeclaredProjectReferences()
+        internal List<ProjectKey> GetDeclaredProjectReferences(bool withoutRefOutputFalse = true)
         {
             var references = _document.XPathSelectElements("//mb:ProjectReference", _manager);
             var list = new List<ProjectKey>();
             var directory = Path.GetDirectoryName(_key.FilePath);
             foreach (var r in references)
             {
+                if (withoutRefOutputFalse)
+                {
+                    // Make sure to check for references that exist only for ordering purposes.  They don't count as 
+                    // actual references.
+                    var refOutputAssembly = r.Element(SharedUtil.MSBuildNamespace.GetName("ReferenceOutputAssembly"));
+                    if (refOutputAssembly != null)
+                    {
+                        bool isRealReference;
+                        if (bool.TryParse(refOutputAssembly.Value.Trim().ToLower(), out isRealReference) && !isRealReference)
```

Comment:
Consider using `out var isRealReference` because it's new and shiny.  Also allows you to collapse to a single `if`.